gdpr/anonymizers/fields.py

```diff
@@ -0,0 +1,246 @@
+            return self.anon_func(value)
+        else:
+            return self.anon_func(self, value)
+
+    def get_is_reversible(self, obj=None):
+        return self.deanonymize_func is not None
+
+    def get_decrypted_value(self, value):
+        if not self.get_is_reversible():
+            raise self.IrreversibleAnonymizationException()
+        else:
+            return self.deanonymize_func(self, value)
+
+
+class DateFieldAnonymizer(NumericFieldAnonymizer):
+    """
+    Anonymization for DateField.
+
+    """
+
+    max_anonymization_range = 365 * 24 * 60 * 60
+
+    def get_encrypted_value(self, value):
+        return value - timedelta(seconds=self.get_numeric_encryption_key())
+
+    def get_decrypted_value(self, value):
+        return value + timedelta(seconds=self.get_numeric_encryption_key())
+
+
+class CharFieldAnonymizer(FieldAnonymizer):
+    """
+    Anonymization for CharField.
+
+    """
+
+    def get_encrypted_value(self, value):
+        return encrypt_message(self.get_encryption_key(), value)
+
+    def get_decrypted_value(self, value):
+        return decrypt_message(self.get_encryption_key(), value)
+
+
+class EmailFieldAnonymizer(FieldAnonymizer):
+
+    def get_encrypted_value(self, value):
+        return encrypt_email(self.get_encryption_key(), value)
+
+    def get_decrypted_value(self, value):
+        return decrypt_email(self.get_encryption_key(), value)
+
+
+class DecimalFieldAnonymizer(NumericFieldAnonymizer):
+    """
+    Anonymization for CharField.
+    """
+
+    max_anonymization_range = 10000
+    decimal_fields: int = 2
+
+    def get_encrypted_value(self, value):
+        return value + Decimal(self.get_numeric_encryption_key()) / Decimal(10 ** self.decimal_fields)
+
+    def get_decrypted_value(self, value):
+        return value - Decimal(self.get_numeric_encryption_key()) / Decimal(10 ** self.decimal_fields)
+
+
+class IPAddressFieldAnonymizer(FieldAnonymizer):
+    """
+    Anonymization for GenericIPAddressField.
+
+    Works for both ipv4 and ipv6.
+    """
+
+    def get_encrypted_value(self, value):
+        return encrypt_ip(self.get_encryption_key(), value)
+
+    def get_decrypted_value(self, value):
+        return decrypt_ip(self.get_encryption_key(), value)
+
+
+AccountNumber = namedtuple('AccountNumber', ['pre_num', 'num', 'bank'])
+
+
+class CzechAccountNumberFieldAnonymizer(FieldAnonymizer):
+    """
+    Anonymization for czech account number.
+    """
+
+    def parse_value(self, value) -> AccountNumber:
+        """
+
+        :param value:
+        :return: (predcisli)-(cislo)/(kod_banky)
+        """
+        account = re.match('(([0-9]{0,6})-)?([0-9]{1,10})/([0-9]{4})', value)
+        return AccountNumber(account[2], account[3], account[4])  # type: ignore
+
+    def get_encrypted_value(self, value):
+        account = self.parse_value(value)
+
+        encrypted_account_num = encrypt_message(self.get_encryption_key(), account.num, NUMBERS)
```

Comment:
pozor číslo účtu má nějaká pravidla, ty budeme muset dodržovat jinak to nepřevedeme na iban. Navíc nám pak selžou validace.

https://www.penize.cz/bezne-ucty/15470-tajemstvi-cisla-uctu